idb/idb.go

```diff
@@ -332,12 +338,27 @@ type AssetBalanceRow struct {
 	Deleted      *bool
 }
 
-// ApplicationRow is metadata relating to one application in an application query.
+// ApplicationRow is metadata and global state (AppParams) relating to one application in an application query.
 type ApplicationRow struct {
 	Application models.Application
 	Error       error
 }
 
+// ApplicationQuery is a parameter object used for query local and global application state.
+type ApplicationQuery struct {
+	Address                  []byte
+	ApplicationID            uint64
+	ApplicationIDGreaterThan uint64
+	IncludeDeleted           bool
+	Limit                    uint64
+}
+
+// AppLocalStateRow is metadata and local state (AppLocalState) relating to one application in an application query.
+type AppLocalStateRow struct {
+	AppLocalState models.ApplicationLocalState
```

Comment:
I really think the database layer shouldn't construct and return REST representation.

Reply:
Using the generated models is how AccountRow and ApplicationRow does it, so I added AppLocalStateRow to match ApplicationRow. However the TxnRow, AssetRow and AssetBalanceRow provide their own encoding. I have no strong feelings either way whether the new type I introduced here should have gone one or the other way ... @winder?